### PR TITLE
bfla: error on unresolved paths

### PR DESCRIPTION
**Pull request by aduminuc**

### CODE REVIEW

File: backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go [hunk 1]
```diff
@@ -55,6 +56,13 @@ func (n *Notifier) NotifyAuthzModel(ctx context.Context, apiID uint, notificatio
 	}); err != nil {
 		return err //nolint:wrapcheck
 	}
+	if log.GetLevel() == log.DebugLevel {
```
Comment (FrimIdan) [inline backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go:59]: better to use `IsLevelEnabled`

File: backend/pkg/modules/internal/bfla/init.go [hunk 2]
```diff
@@ -325,11 +325,15 @@ func (h httpHandler) GetEvent(w http.ResponseWriter, r *http.Request, eventID in
...
 		log.Warnf("tags for apiID=%d not found", event.APIInfoID)
 	}
-	resolvedPath := bfladetector.ResolvePath(tags, event)
-	if obj, err := h.bflaDetector.FindSourceObj(resolvedPath, string(event.Method), src.Uid, event.APIInfoID); err != nil {
-		log.Error(err)
-	} else if !obj.Authorized {
-		e.BflaStatus = bfladetector.ResolveBFLAStatusInt(int(event.StatusCode))
+	resolvedPath, err := bfladetector.ResolvePath(tags, event)
+	if err != nil {
+		log.Warnf("%v", err)
```
Comment (FrimIdan) [inline backend/pkg/modules/internal/bfla/init.go:330]: better to add more informative error message

File: backend/pkg/modules/internal/bfla/init.go [hunk 3]
```diff
@@ -325,11 +325,15 @@ func (h httpHandler) GetEvent(w http.ResponseWriter, r *http.Request, eventID in
...
+	} else {
+		if obj, err := h.bflaDetector.FindSourceObj(resolvedPath, string(event.Method), src.Uid, event.APIInfoID); err != nil {
+			log.Error(err)
```
Comment (FrimIdan) [inline backend/pkg/modules/internal/bfla/init.go:333]: better to add more informative error message

File: backend/pkg/modules/internal/bfla/init.go [hunk 4]
```diff
@@ -659,18 +663,22 @@ func (h httpHandler) PutEventIdOperation(w http.ResponseWriter, r *http.Request,
...
-			h.bflaDetector.ApproveTrace(resolvedPath, string(apiEvent.Method), src, apiEvent.APIInfoID, nil)
-		case restapi.Deny:
-			h.bflaDetector.DenyTrace(resolvedPath, string(apiEvent.Method), src, apiEvent.APIInfoID, nil)
-		case restapi.ApproveUser:
-			h.bflaDetector.ApproveTrace(resolvedPath, string(apiEvent.Method), src, apiEvent.APIInfoID, user)
-		case restapi.DenyUser:
-			h.bflaDetector.DenyTrace(resolvedPath, string(apiEvent.Method), src, apiEvent.APIInfoID, user)
+		resolvedPath, err := bfladetector.ResolvePath(tags, apiEvent)
+		if err != nil {
+			log.Warnf("%v", err)
```
Comment (FrimIdan) [inline backend/pkg/modules/internal/bfla/init.go:668]: better to add more informative error message

File: backend/pkg/modules/internal/bfla/init.go [hunk 5]
```diff
@@ -325,11 +325,15 @@ func (h httpHandler) GetEvent(w http.ResponseWriter, r *http.Request, eventID in
...
-	if obj, err := h.bflaDetector.FindSourceObj(resolvedPath, string(event.Method), src.Uid, event.APIInfoID); err != nil {
-		log.Error(err)
-	} else if !obj.Authorized {
-		e.BflaStatus = bfladetector.ResolveBFLAStatusInt(int(event.StatusCode))
+	resolvedPath, err := bfladetector.ResolvePath(tags, event)
+	if err != nil {
+		log.Warnf("path not found in the spec: %v", err)
+	} else {
+		if obj, err := h.bflaDetector.FindSourceObj(resolvedPath, string(event.Method), src.Uid, event.APIInfoID); err != nil {
+			log.Error("unable to map the source onto an existing entity in the auth model", err)
```
Comment (FrimIdan) [inline backend/pkg/modules/internal/bfla/init.go:333]: use `log.Errorf` and add `: %v` at the end to print the error

File: backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go [hunk 6]
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/openclarity/apiclarity/api3/global"
 	"github.com/openclarity/apiclarity/api3/notifications"
 	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/core"
+	log "github.com/sirupsen/logrus"
```
Comment (klyr) [inline backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go:24]: go imports

File: backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go [hunk 7]
```diff
@@ -55,6 +56,13 @@ func (n *Notifier) NotifyAuthzModel(ctx context.Context, apiID uint, notificatio
 	}); err != nil {
 		return err //nolint:wrapcheck
 	}
+	if log.IsLevelEnabled(log.DebugLevel) {
+		jntf, err := ntf.MarshalJSON()
+		if err == nil {
+			log.Debugf("Auth model notification: %s", jntf)
+
```
Comment (klyr) [inline backend/pkg/modules/internal/bfla/bfladetector/bfla_notifier.go:63]: gofmt :)

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 8]
```diff
@@ -522,38 +527,67 @@ func (l *learnAndDetectBFLA) commandsRunner(ctx context.Context, command Command
...
+					nsMap, ok := audienceMap[aud.K8sObject.Name]
+					if !ok {
+						return oldModel, fmt.Errorf("unable to find the audience entry for %v, %v, name = %s, namespace = %s: name not found", op.Path, op.Method, aud.K8sObject.Name, aud.K8sObject.Namespace)
+					}
+					foundAud, ok := nsMap[aud.K8sObject.Namespace]
+					if !ok {
+						if aud.K8sObject.Namespace == "" && len(nsMap) == 1 {
+							for _, foundAud = range nsMap {
+								break
+							}
```
Comment (klyr) [inline backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:581]: I don't get it

Comment (aduminuc) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:581]: Get the first (and only) element in the map .

**Review by klyr — APPROVED** [review]
LGTM

**Pull request merged**